botnim/vector_store/vector_store_es.py

```diff
@@ -48,8 +51,8 @@ def __init__(self, config, config_dir, es_host, es_username, es_password,
 
     def _index_name_for_context(self, context_name: str) -> str:
         """Standardize index name construction"""
-        base_name = f"{self.config['slug']}__{context_name}"
-        return self.env_name_slug(base_name.lower().replace(' ', '_'))
+        environment = 'production' if self.production else 'staging'
```

Comment:
If we need to recreate the name here then it's a sign that `get_index_name` should receive a `production` boolean and not an `environment` string

Reply:
fixed. notice I had to handle the vector store initialization state between contexts.